e2e/cypress/e2e/spec.cy.js

```diff
@@ -77,6 +77,78 @@ describe("journey test", () => {
     cy.findByText("All Day").should("exist");
   });
 
+  it("shows correct default time when uncreated event is changed to not be `allDay`", () => {
+    cy.visit("http://localhost:3000");
+    cy.intercept({
+      method: "POST",
+      url: "/entries",
+      hostname: "localhost",
+    }).as("createEntry");
+
+    // Creating new event from "+"
+    cy.get(`[aria-label="add event"]`).click();
+    cy.contains("label", "Title").click().type("Bye");
+    cy.contains("label", "Description").click().type("It's a beautiful night");
+    cy.contains("label", "Start Date").click().type("2022-12-14");
+    cy.contains("label", "End Date").click().type("2022-12-14");
+    cy.contains("label", "All Day").click();
+
+    cy.contains("label", "Start Time").should("not.exist");
+    cy.contains("label", "End Time").should("not.exist");
+
+    cy.contains("label", "All Day").click();
+
+    // Create event modal has correct time defaults
+    cy.get(`[id="startTime"]`).should("have.value", "04:00");
+    cy.get(`[id="endTime"]`).should("have.value", "05:00");
+    cy.contains("button", "Create Event").click();
+  });
+
+  it("shows correct default time when new and existing allDay event is changed to not be `allDay`", () => {
+    cy.visit("http://localhost:3000");
+    cy.intercept({
+      method: "POST",
+      url: "/entries",
+      hostname: "localhost",
+    }).as("createEntry");
+
+    // Creating new event from clicking on a date
+    cy.get(`[data-date="2022-12-14"]`).click();
+    cy.contains("label", "Title").click().type("Bye");
+    cy.contains("label", "Description").click().type("It's a beautiful night");
+    cy.contains("label", "Start Date").click().type("2022-12-14");
+    cy.contains("label", "End Date").click().type("2022-12-14");
+
+    // Assert All Day is default true
+    cy.contains("label", "All Day").within(() => {
+      cy.get('[type="checkbox"]').should("be.checked");
+    });
+
+    // Uncheck All Day
+    // Assert time defaults are correct
+    cy.contains("label", "All Day").click();
+    cy.get(`[id="startTime"]`).should("have.value", "04:00");
+    cy.get(`[id="endTime"]`).should("have.value", "05:00");
+
+    // Recheck All Day to create event
+    cy.contains("label", "All Day").click();
+    cy.contains("button", "Create Event").click();
+    cy.wait("@createEntry").then((interception) => {
+      postTwoId = interception.response.body._id;
```

Comment:
postTwoId needs to be a different variable. Since we create a new post in this test, we should probably create a new variable to keep track of this id to do the deletion cleanup.

Reply:
Updated!